src/pymovements/utils/parsing.py

```diff
@@ -52,96 +53,146 @@ def check_nan(sample_location: str) -> float:
     return ret
 
 
+def compile_patterns(patterns: list[dict[str, Any]]) -> list[dict[str, Any]]:
+    """Compile patterns from strings.
+
+    Parameters
+    ----------
+    patterns:
+        The list of patterns to compile.
+    """
+    msg_prefix = r'MSG\s+\d+\s+'
+
+    compiled_patterns = []
+
+    for pattern in patterns:
+        if isinstance(pattern, str):
+            compiled_pattern = {'pattern': re.compile(msg_prefix + pattern)}
+            compiled_patterns.append(compiled_pattern)
+            continue
+
+        if isinstance(pattern, dict):
+            if isinstance(pattern['pattern'], str):
+                compiled_patterns.append({
+                    **pattern,
+                    'pattern': re.compile(msg_prefix + pattern['pattern']),
+                })
+                continue
+
+            if isinstance(pattern['pattern'], tuple):
+                for single_pattern in pattern['pattern']:
+                    compiled_patterns.append({
+                        **pattern,
+                        'pattern': re.compile(msg_prefix + single_pattern),
+                    })
+                continue
+
+            raise ValueError(f'invalid pattern: {pattern}')
+
+        raise ValueError(f'invalid pattern: {pattern}')
+
+    return compiled_patterns
+
+
+def get_additional_columns(compiled_patterns: list[dict[str, Any]]) -> set[str]:
+    """Get additionally needed columns from compiled patterns."""
+    additional_columns = set()
+
+    for compiled_pattern_dict in compiled_patterns:
+        if 'column' in compiled_pattern_dict:
+            additional_columns.add(compiled_pattern_dict['column'])
+
+        for column in compiled_pattern_dict['pattern'].groupindex.keys():
+            additional_columns.add(column)
+
+    return additional_columns
+
+
 def parse_eyelink(
         filepath: Path,
-        sync_msg_start_pattern: None | str = None,
-        sync_msg_stop_pattern: None | str = None,
+        patterns: list | None = None,
+        schema: dict | None = None,
```

Comment:
note that `list`/ `dict` is not a helpful type since we do not know what the list contains.

Reply:
thanks a lot for taking care of this!